src/pages/on-chain-governance/index.page.tsx

```diff
@@ -98,245 +109,425 @@ export default function OnChainGovernancePage(props) {
+        <Button
+          testId="dummy-proposal"
+          label="Create dummy proposal"
+          onClick={createDummyProposals}
+          customStyle="hover:bg-gray-50"
+        />
+      )}
+
+      <Container className="md:pt-11 pt-10 pb-20">
+        <div className="flex md:flex-row flex-col">
+          <div className="flex flex-col grow">
+            {/* main title */}
             <div
-              data-testid="OnChainGovernance.ProposalsSubmittedTitle"
-              className="text-gray-500 text-lg"
+              data-testid="OnChainGovernance.Title"
+              className="text-[10px] tracking-[0.0015em] font-medium text-gray-500 dark:text-dark-gray-900"
             >
-              {OnChainGovernanceTitles.proposalsSubmittedTitle}
+              ON-CHAIN GOVERNANCE
             </div>
             <div
-              data-testid="OnChainGovernance.ProposalsSubmitted"
-              className="font-semibold text-gray-900 text-lg"
+              data-testid="OnChainGovernance.Proposals.Title"
+              className="text-4xl leading-[48px] tracking-[0.0015em] font-semibold dark:text-dark-gray-900"
             >
-              {votingCycle.proposalsSubmitted}
+              Proposals
             </div>
+
+            {/* Set Masternode */}
+            {isMasterNodeClicked ? (
+              <>
+                <div className="flex flex-row items-center gap-x-[10px] mt-2">
+                  <div
+                    onBlur={() => {
+                      setIsMasterNodeInputFocus(false);
+                    }}
+                    onFocus={() => {
+                      setIsMasterNodeInputFocus(true);
+                    }}
+                    className={classNames(
+                      "flex flex-row rounded border py-2 px-4 lg:w-[385px] md:w-[190px] dark:bg-gray-800",
+                      { "border-primary-300": isMasterNodeInputFocus },
+                      { "border-red-200": masterNodeErrorMsg !== "" }
+                    )}
+                  >
+                    <input
+                      onChange={(v) => setMasterNodeID(v.target.value)}
+                      value={masterNodeID}
+                      className="w-2/3 text-sm focus:outline-none grow focus:caret-[#007AFF] dark:bg-gray-800 dark:text-dark-gray-900"
+                      placeholder="Set your masternode"
+                    />
+                    {(masterNodeID !== "" || isMasterNodeInputFocus) && (
+                      <MdClose
+                        onClick={() => {
+                          setMasterNodeID("");
+                        }}
+                        size={15}
+                        className="text-gray-500 self-center cursor-pointer m-auto"
+                      />
+                    )}
+                  </div>
+                  <Button
+                    label="SAVE"
+                    testId="OnChainGovernance.SaveMasterNodeID"
+                    disabled={masterNodeID === ""}
```

Comment:
should disabled when input validation returns error or false

Reply:
updated to disable Save button when Invalid masternode address